clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java

```diff
@@ -367,19 +395,25 @@ public void initialize(URI uri, Configuration conf) throws IOException {
       mStatistics.incrementReadOps(1);
     }
 
-    if (!mTFS.exist(tPath)) {
+    List<FileInfo> files;
+    try {
+      TachyonFile file = mTFS.open(tPath);
+      files = mTFS.listStatus(file);
+    } catch (InvalidPathException e) {
       throw new FileNotFoundException("File does not exist: " + path);
```

Comment:
Are we committed to moving to Java 7? If so we can use (InvalidPathException | FileDoesNotExistException e) instead